internal/graph/check_test.go

```diff
@@ -0,0 +1,72 @@
+package graph
+
+import (
+	"context"
+	"testing"
+
+	parser "github.com/craigpastro/openfga-dsl-parser/v2"
+	"github.com/oklog/ulid/v2"
+	"github.com/openfga/openfga/pkg/storage/memory"
+	"github.com/openfga/openfga/pkg/tuple"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/stretchr/testify/require"
+	openfgav1 "go.buf.build/openfga/go/openfga/api/openfga/v1"
+)
+
+func TestResolveCheckDeterministic(t *testing.T) {
```

Comment:
Do we need a test for union (the test right now is intersection)

Reply:
Done. Added a condition on `define viewer: [group#member] as self or editor` and added some tuples that generate a model resolution too complex path on `document:1#editor`, and yet the test still passes with the expected outcome like it should. So this proves union is also deterministic in regards to an `{allowed: true}` outcome superseding a resolution error.